istioctl/pkg/precheck/precheck.go

```diff
@@ -179,6 +187,66 @@ func checkExternalNameAlias(cli kube.CLIClient, messages *diag.Messages) error {
 	return nil
 }
 
+func checkPilot(cli kube.CLIClient, namespace string, messages *diag.Messages) error {
+	deployments, err := cli.Kube().AppsV1().Deployments(namespace).List(context.TODO(), metav1.ListOptions{
+		LabelSelector: "app=istiod",
+	})
+	if err != nil {
+		return err
+	}
+	for _, deployment := range deployments.Items {
+		scopingImpacted := false
+
+		// Obtain configmap to verify if affected features are used
+		configMapName := "istio"
+		if rev := deployment.Labels[label.IoIstioRev.Name]; rev != "default" {
+			configMapName += fmt.Sprintf("-%s", rev)
+		}
+		configMap, err := cli.Kube().CoreV1().ConfigMaps(namespace).Get(context.TODO(), configMapName, metav1.GetOptions{})
+		if err != nil {
+			fmt.Printf("Error getting configmap %s: %v\n", configMapName, err)
+		}
+		meshData := make(map[string]interface{})
+		if data, exists := configMap.Data["mesh"]; exists {
+			if err := yaml.Unmarshal([]byte(data), &meshData); err != nil {
+				fmt.Printf("Error parsing meshConfig: %v\n", err)
+				return err
+			}
+		}
+		scopingImpacted = meshData["discoverySelectors"] != nil
+
+		// Check if mitigation is already in place
+		for _, container := range deployment.Spec.Template.Spec.Containers {
+			if container.Name == "discovery" {
+				for _, envVar := range container.Env {
+					if envVar.Name == "ENHANCED_RESOURCE_SCOPING" && envVar.Value == "true" {
+						scopingImpacted = false
```

Comment:
can be optimized. We donot need to check env when meshData["discoverySelectors"] == nil

Reply:
Added early skip logic